### PR TITLE
Add Pylint checks

### PR DESCRIPTION
**Pull request by csc-felipe**
### Description

Using Pylint to check the code makes us a bit less prone to introduce technical debts unknowingly in the code.

Some checks have been disabled, and we may consider enabling and fixing some of them in the future.

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Changes Made

- Add checks with pylint
- Fix many pylint complaints
- Removed pydocstyle, which is already integrated into flake8
- I didn't see the benefit of `doc8` at this point, so removed it

### Testing

- [x] Tests do not apply

### CODE REVIEW

**Review by csc-jm — APPROVED** [review]
Initially I thought the list of checks in the pylint config file was a bit excessive but they only seem to take about 5 sec on my machine so it's fine. Otherwise I'm also fine with all the things it changes in this PR

File: metadata_backend/api/operators.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# pylint: disable=too-many-lines
```
[the remark this answers is not in the document]
Comment (csc-felipe) [reply metadata_backend/api/operators.py:1]: My idea to leave it like this is exactly that it would be a good idea to split it up.

**Comment by csc-felipe** [comment]
I used the automatic configuration generator, so it explicitly shows all the enabled checks. What I did was enable all and choose which ones are disabled. And then added the `docparams` extension so that it checks the function parameters match the docstrings.

File: metadata_backend/api/auth.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ async def login(self, req: Request) -> Response:
         # Redirect user to AAI
         response = web.HTTPSeeOther(session["url"])
         response.headers["Location"] = session["url"]
-        raise response
+        return response
```
Comment (blankdots) [inline metadata_backend/api/auth.py:69]: @csc-felipe i think this should have been left as a `raise`, have you tested this with an odic provider ?

this is something we cannot (or it is very difficult) mock with the integration tests, and needs to be tested with a oidc provider

Comment (csc-felipe) [reply metadata_backend/api/auth.py:69]: The current implementation works with SDS AAI, I have tried it manually, but what kind of tests did you have in mind?

The difference between `raise` and `return` is not so dramatic. afaik, it's more about how the middleware handles them than anything else.

Comment (csc-felipe) [reply metadata_backend/api/auth.py:69]: https://github.com/CSCfi/metadata-submitter/pull/559

**Pull request merged**